distributed/protocol/tests/test_torch.py

```diff
@@ -31,3 +31,10 @@ def test_resnet():
     header, frames = serialize(model)
     model2 = deserialize(header, frames)
     assert str(model) == str(model2)
+
+
+def test_deserialize_grad():
+    torch.set_printoptions(precision=10)
+    a = np.random.rand(8, 1)
+    t = torch.tensor(a, requires_grad=True, dtype=torch.float)
+    deserialize(*serialize(t))
```

Comment:
The result of this line should be a PyTorch tensor that is mostly equivalent to `t`.  Is there some check we can run to verify that we have recreated an equivalent tensor object?

Reply:
By doing this: 

```
import torch
import numpy as np
from distributed.protocol import serialize, deserialize

a = np.random.rand(8, 1)

t = torch.tensor(a, requires_grad=True, dtype=torch.float)

t2 = deserialize(*serialize(t))
print(a)
print(t2.numpy())

```

I get: 
```
[[0.90611823]
 [0.57207592]
 [0.14868348]
 [0.24763735]
 [0.25475143]
 [0.71949837]
 [0.19011521]
 [0.3211044 ]]
[[0.9061182 ]
 [0.5720759 ]
 [0.14868347]
 [0.24763735]
 [0.25475144]
 [0.7194984 ]
 [0.19011521]
 [0.3211044 ]]
```

They slightly differ on the 8th decimal. `assert` will thorw an error because of that and I thought that setting larger precission would help.

Reply:
Getting to check if they are equivalent, I could use https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.around.html#numpy.around. And truncate to the 7th or 6th decimal.

Reply:
> They slightly differ on the 8th decimal

I use `np.allclose` in these situations.

Reply:
@stsievert that worked. Thanks for the suggestion.